docs/proposals/connector-protocol-v2.md

```diff
@@ -224,58 +226,53 @@ notify(StreamId, StreamName, PointOfRef) ->
         point_of_reference(PointOfRef)
     ]).
 
--spec message(Message, MessageId, StreamId, Flags) -> frame() when
-    Message :: boundary | {non_neg_integer(), iodata()} | iodata(),
-    MessageId :: non_neg_integer(),
+-spec message(Flags, StreamId, MessageId, EventTime, Key, Message) -> frame()
+  when
+    Flags :: non_neg_integer(),      % See message bit flag definitions above
     StreamId :: non_neg_integer(),
-    Flags :: non_neg_integer(). % See message bit flag definitions above
-message(_Message=boundary, MessageId, StreamId, Flags) ->
-    frame([
-        ?MESSAGE,
-        u64(StreamId),
-        u16(Flags bor ?BOUNDARY),
-        u64(MessageId)
-    ]);
-message({EventTime, Message}, MessageId, StreamId, Flags) ->
-    frame([
-        ?MESSAGE,
-        u64(StreamId),
-        u16(Flags bor ?EVENT_TIME),
-        u64(MessageId),
-        u64(EventTime),
-        Message
-    ]);
-message(Message, MessageId, StreamId, Flags) ->
+    MessageId :: non_neg_integer(),
+    EventTime :: non_neg_integer(),
+    Key :: string(),
+    Message :: iodata().
```

Comment:
We may want to put metadata into a map. Flags, MessageId, EventTime, Key, and Message itself all seem like things worth grouping for readability sake (it's a shame erlang can't spec map values but we could pretend in this case).

```erlang
message(StreamId, IsEphemeral, Message) -> ...
  % Note that we may not need ephemeral for the first pass so we might be better off dropping it
  % until later?

make_message(MessageId, EventTime, Key, Value) ->
  % explicit construction
  Flags = value_to_flag(EventTime, ?EVENT_TIME) band value_to_flag(Key, ?KEY) band value_to_flag(Value, ?BOUNDARY)
  #{message_id => MessageId, event_time => EventTime, key => Key, value => Value, flags => Flags}.

value_to_flag(none, _) -> 0;
value_to_flag(_, Flag) -> Flag.
```

Just a thought if we're maintaining this spec.